tephra-core/src/main/java/co/cask/tephra/AbstractTransactionAwareTable.java

```diff
@@ -70,8 +70,10 @@ public void startTx(Transaction tx) {
   @Override
   public Collection<byte[]> getTxChanges() {
     Collection<byte[]> txChanges = new TreeSet<byte[]>(UnsignedBytes.lexicographicalComparator());
-    for (ActionChange change : changeSet) {
-      txChanges.add(getChangeKey(change.getRow(), change.getFamily(), change.getQualifier()));
+    if (conflictLevel != TxConstants.ConflictDetection.NONE) {
```

Comment:
Shall we check for `NONE` first and return `Collections.emptyList()` to avoid the overhead of creating new TreeSet?